### PR TITLE
feat(lambda-promtail): add cloudtrail log ingestion support

### PR DESCRIPTION
**Pull request by CCOLLOT**
**What this PR does / why we need it**:
### Add support for AWS CloudTrail audit logs ingestion using lambda-promtail

Calls to AWS APIs are logged in AWS Cloudtrail and are helpful for security and debugging purposes. However, I've experienced difficulties with it:
+ The AWS CloudTrail service is not well integrated with Prometheus (no metrics, no alerts) and I don't want to manage alerts in CloudWatch Alerts
+ The search experience is painful with CloudTrail via the AWS Console (I will not elaborate 😅).

This PR allows ingesting CloudTrail audit logs sent to an S3 bucket using the same approach as VPC flow logs or Load Balancer logs.

**Special notes for your reviewer**:

+ Because the Cloudtrail file format is not text but json, we stream the json CloudTrail records instead of using the already existing scanner.
+ Because the Cloudtrail filename format is not the same as for the Flow log or the Load balancer log files, we need to split the regexes by service (although many AWS services seem to share the same `defaultFilenameRegex`).
+ In the `getLabels` function, we expect the `type` parameter to be found in the filename using the Regex. For some log files (ex: Cloudfront log files). The file name has no reference to the service name.  This is why, as a default, when no type is found in the name of the file, I set it to use the key of the matching Regex expression.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

File: tools/lambda-promtail/lambda-promtail/s3.go [hunk 1]
```diff
@@ -65,23 +76,50 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
...
+	logType := ""
+	switch labels["type"] {
+	case FLOW_LOG_TYPE:
 		skipHeader = true
-		logType = "s3_vpc_flow"
-	} else if labels["type"] == LB_LOG_TYPE {
+		logType = "s3_vpc_flow"	
+	case LB_LOG_TYPE:
 		logType = "s3_lb"
+    case CLOUDTRAIL_LOG_TYPE:
```
Comment (MichelHollands) [inline tools/lambda-promtail/lambda-promtail/s3.go:86]: Is this correctly formatted? Can you run `go fmt` on it?

File: tools/lambda-promtail/lambda-promtail/s3.go [hunk 2]
```diff
@@ -113,21 +151,30 @@ func parseS3Log(ctx context.Context, b *batch, labels map[string]string, obj io.
...
-
-	match := filenameRegex.FindStringSubmatch(labels["key"])
-	for i, name := range filenameRegex.SubexpNames() {
+	var matchingExp *regexp.Regexp
+	var matchingType *string
+	for key, exp := range filenameRegexes {
+		if exp.MatchString(labels["key"]) {
+			matchingExp = exp
+			matchingType = aws.String(key)
+		} else {
```
[the remark this answers is not in the document]
Comment (CCOLLOT) [reply tools/lambda-promtail/lambda-promtail/s3.go]: Nice catch, I forgot to remove it 👍

**Comment by CCOLLOT** [comment]
> @CCOLLOT Please also fix the merge conflicts.

There you go 👍

**Review by MichelHollands — APPROVED** [review]

**Pull request merged**